test/e2e-v2/cases/satellite/native-protocols/expected/dependency-services.yml

```diff
@@ -14,35 +14,18 @@
 # limitations under the License.
 
 nodes:
-{{- range .nodes }}
```

Comment:
FYI @mrproliu @wankai123 @dmsolr @arugal , `range` + `if` is not designed to verify lists containing specific elements, can you please review other repos E2E and see whether this style is used, if so, please also update them?

For example, the following rules expect there are 4 nodes, but it still passed even if there is only one node (one arbitrary node in the 4 listed):

```yaml
nodes:
{{- range .nodes }}
{{- if eq .name "User" }}
- id: {{ b64enc "User" }}.0
  name: User
  type: USER
  isreal: false
{{- end }}
{{- if eq .name "e2e-service-provider" }}
- id: {{ b64enc "e2e-service-provider"}}.1
  name: e2e-service-provider
  type: Tomcat
  isreal: true
{{- end }}
{{- if eq .name "e2e-service-consumer" }}
- id: {{ b64enc "e2e-service-consumer"}}.1
  name: e2e-service-consumer
  type: Tomcat
  isreal: true
{{- end }}
{{- if eq .name "localhost:-1" }}
- id: {{ b64enc "localhost:-1" }}.0
  name: localhost:-1
  type: H2
  isreal: false
{{- end }}
{{- end }}
```

Reply:
What is the recommended way? Seems should separate to verify one node per range?

Reply:
What is the recommended way? Seems should separate to verify one node per range?

Reply:
> What is the recommended way? Seems should separate to verify one node per range?

Check what I updated in this patch. `contains` is designed to do this.

Reply:
@mrproliu @wankai123 @arugal You should have other cases having the same issue.